### ISSUE TITLE
RCS1102  C# Consider marking class as static

### ISSUE THREAD

**Issue by cspwizard** [JosefPihrt/Roslynator#23]
I've false errors RCS1102  C# Consider marking class as static. in project for classes like

```C#
internal abstract class Base
{
   // here the RCS1102  error  occurs
   internal abstract class Const  
   {
   }
}
```
and derived from it 

```C#
internal class Derived : Base
{
   // here the RCS1102  error occurs
   internal new class Const : Base.Const
   {
   }
}
```

According to C# rules abstract classes cannot be static, and static classes should be inherited from the Object. 
So the check fails in both cases

**Comment by JosefPihrt** [comment]
Thanks for the report @cspwizard !
